draft-nygren-httpbis-httpssvc.md

```diff
@@ -0,0 +1,791 @@
+
+The SvcDomainName MUST point to a domain name that contains
+another HTTPSSVC record and/or address (AAAA and/or A) records.
+
+Note that the RRName or the SvcDomainName MAY themselves be CNAMEs.
+Clients and recursive resolvers MUST follow CNAMEs as normal.
+
+Due to the risk of loops, clients and recursive resolvers MUST
+implement loop detection.  Chains of consecutive HTTPSSVC and CNAME
+records SHOULD be limited to (8?) prior to reaching terminal address
+records.
+
+The SvcFieldValue in this form SHOULD be an empty string and
+clients MUST ignore its contents.
+
+
+## HTTPSSVC records: alternative service form
+
+When SvcRecordType is 1, the combination of SvcDomainName and
+SvcFieldValue within each resource record associates an Alternative
+Service Field Value with an origin.
+
+The SvcFieldValue of the HTTPSSVC resource record contains an Alt-Svc
+Field Value, exactly as defined in Section 4 of {{!AltSvc}},
+but with the uri-host moved to the SvcDomainName field.
+
+For example, if the operator of https://www.example.com
+intends to include an HTTP response header like
+
+ Alt-Svc: hq="svc.example.net:8003"; ma=3600, \
+          h2="svc.example.net:8002"; ma=3600
+
+They would also publish an HTTPSSVC DNS RRSet like
+
+ www.example.com. 3600 IN HTTPSSVC 1 svc.example.net. "hq=\\":8003\\" pri=0"
+                          HTTPSSVC 1 svc.example.net. "h2=\\":8002\\" pri=1"
+
+This data type can be represented as an Unknown RR as described in
+{{!RFC3597}}:
+
+ www.example.com. 3600 IN TYPE??? \\# TBD:WRITEME
+
+This construction is intended to be extensible in two ways.  First,
+any extensions that are made to the Alt-Svc format for transmission
+over HTTPS are also applicable here, unless expressly mentioned
+otherwise.
+
+Second, by defining a way to map non-HTTPS schemes and non-default
+ports we provide a way for the HTTPSSVC to be used for them as needed.
+However, by using the origin name for the RRName for scheme https and
+port 443 we allow HTTPSSVC records to be included at the end of CNAME
+chains for existing site implementations without requiring changes in
+the zone containing the origin.
+
+
+# Differences from Alt-Svc as transmitted over HTTP
+
+Publishing an alternative services form HTTPSSVC record in DNS is
+intended to be equivalent to transmitting this field value over HTTPS,
+and receiving an HTTPSSVC record is intended to be equivalent to
+receiving this field value over HTTPS.  However, there are some small
+differences in the intended client and server behavior.
+
+## Omitting Max Age and Persist
+
+When publishing an HTTPSSVC record in DNS, server operators MUST omit the
+"ma" parameter, which encodes the "max age" (i.e. expiration time) of
+an Alt-Svc Field Value.  Instead, server operators SHOULD encode the
+expiration time in the DNS TTL, and MUST NOT set a TTL longer than the
+intended "max age".
+
+When receiving an HTTPSSVC record, clients MAY synthesize a new "ma"
```

Comment:
Why are clients only permitted to synthesize a lifetime instead of encouraged?

Reply:
I agree, how about
> clients SHOULD synthesize a new "ma" parameter from the DNS TTL if they intend to integrate with an Alt-Svc cache.